Sage/carbon
# feat(tab): introduce titleProps for passing props to corresponding title

### Proposed behaviour

- `Tab`: Add new `titleProps` prop - which accepts an object of props to pass to the tab's corresponding `TabTitle`.
- `TabTitle`: Accepts new `data-role` prop to be used for testing and analytics purposes

### Current behaviour

- Consumers cannot set a `data-role` property for a `Tab`'s corresponding `TabTitle`, which has been raised as a feature request to allow consumers to query a tab's title using testing/analytics tools.

### Checklist




### Testing instructions

- Verify tab title has `data-role` property set when specified
- Double check typescript and docs definitions 

The following CodeSandbox is an example of the broken behaviour: [Go to codesandbox](https://codesandbox.io/s/tab-data-role-6v5wvp?file=/src/App.js)
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
